### ISSUE TITLE
Unexpected website error when using idSite=all

### ISSUE THREAD

**Issue by gricey432** [matomo-org/matomo#12312]
Using the API
```
https://example.com/?module=API&method=Live.getLastVisitsDetails&idSite=all&period=day&date=today&format=json&token_auth=xxx
```

Should return the latest visit details for all sites on the server, instead I get
```
{
result: "error",
message: "An unexpected website was found in the request: website id was set to '0' ."
}
```

**Comment by gricey432** [comment]
Additionally if I use `idSite=1,2,3` I only get the data for `idSite=1`. I have a feeling this issue is only on the Live endpoint though, not sure..

**Comment by tassoman** [comment]
I bet idSite must be a single integer. You're asking the _getLastVisitsDetails_ method in the _Live_ plugin. If you want all sites results, you must query _Multisite_ plugin.

There's an [example on demo site](https://demo.piwik.org/index.php?module=API&action=listAllAPI&idSite=3&period=day&date=yesterday#MultiSites)

**Comment by gricey432** [comment]
So this is expected behaviour?

The Standard API parameters section of the [docs](https://developer.piwik.org/api-reference/reporting-api) says `all` and `1,4,5,6` are valid values, and the [Live!](https://developer.piwik.org/api-reference/reporting-api#Live) section doesn't mention otherwise.

**Comment by Findus23** [comment]
@gricey432 Did you test this in the latest version?

I think #12854 should have solved this issue.

**Comment by gricey432** [comment]
Just tested and this appears to be fixed!

Resolved in #12854
